src/z3c/checkversions/main.py

```diff
@@ -82,17 +82,15 @@ def main():
 
     if buildoutcfg:
         from . import buildout
-        checker = buildout.Checker(filename=buildoutcfg,
-                                   blacklist=options.blacklist,
-                                   incremental=options.incremental,
-                                   verbose=options.verbose,
-                                   **kw)
+        factory = buildout.Checker
     else:
         from . import installed
-        checker = installed.Checker(blacklist=options.blacklist,
-                                    incremental=options.incremental,
-                                    verbose=options.verbose)
+        factory = installed.Checker
 
+    checker = factory(blacklist=options.blacklist,
+                      incremental=options.incremental,
+                      verbose=options.verbose,
+                      **kw)
```

Comment:
You're no longer passing `filename` to `buildut.Checker`?  How do things work, given
https://github.com/zopefoundation/z3c.checkversions/blob/be59ba7b131b0773e10018abdccf6a40a545659e/src/z3c/checkversions/buildout.py#L29-L33

Is the code untested?

Reply:
Good catch @mgedmin, I fixed passing the filename and added tests for the main() function in b77bcf803fae2bad387ce2ba5698f20e6cec7b04